### PR TITLE
Implement tracing in action server

### PR DESCRIPTION
**Pull request by Tawakalt**
**Proposed changes**:
- Fixes ATO-783

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation in the [rasaHQ/rasa](https://github.com/rasaHQ/rasa)
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa-sdk/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa-sdk#code-style) for instructions)

### CODE REVIEW

**Comment by Tawakalt** [comment]
@vcidst Asking for first round of review while I add tests 🙏🏾

File: tests/tracing/test_config.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+from tests.tracing import conftest
+from tests.tracing.conftest import (
+    TRACING_TESTS_FIXTURES_DIRECTORY,
+)
+
+UDP_BUFFER_SIZE = 2048
+
+
+def test_jaeger_config_correctly_extracted() -> None:
+    cfg = EndpointConfig(
```
Comment (vcidst) [inline tests/tracing/test_config.py:16]: Can you please add a small comment on these tests about what they're testing?

File: pyproject.toml [hunk 2]
```diff
@@ -83,6 +83,10 @@ prompt-toolkit = "^3.0,<3.0.29"
 "ruamel.yaml" = ">=0.16.5,<0.18.0"
 websockets = ">=10.0,<12.0"
 pluggy = "^1.0.0"
+opentelemetry-api = "~1.15.0"
+opentelemetry-sdk = "~1.15.0"
+opentelemetry-exporter-jaeger = "~1.15.0"
+opentelemetry-exporter-otlp = "~1.15.0"
```
[the remark this answers is not in the document]
Comment (Tawakalt) [reply pyproject.toml:89]: Not really. Just for consistency sake I guess 🤷🏾‍♀️

File: tests/tracing/fixtures/otlp_endpoints.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+tracing:
+  type: otlp
+  endpoint: localhost:4317
+  insecure: true
```
[the remark this answers is not in the document]
Comment (Tawakalt) [reply tests/tracing/fixtures/otlp_endpoints.yml:4]: I've added a test for that

**Comment by Tawakalt** [comment]
> We should add a section in the readme about what-is-tracing and how to add any additional spans, span attributes if someone wants to (say span inside custom actions or helper methods)

I'd prefer that documentation to be a separate ticket so it's not blocking this.

File: changelog/1016.feature.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Implemented tracing in the action server.
```
Comment (vcidst) [inline changelog/1016.feature.md:1]:
```suggestion
Adds tracing functionality to the Rasa SDK, bringing enhanced monitoring, execution profiling and debugging capabilities to the Rasa Actions Server. See Rasa Documentation on Tracing to know more about adding spans and span attributes to your custom actions
```

File: rasa_sdk/tracing/utils.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+
+def get_tracer_provider(
+    cmdline_arguments: argparse.Namespace,
+) -> Optional[TracerProvider]:
+    """Gets the tracer provider from the command line arguments."""
+    tracer_provider = None
+    endpoints_file = ""
+    if "endpoints" in cmdline_arguments:
+        endpoints_file = cmdline_arguments.endpoints
```
Comment (vcidst) [inline rasa_sdk/tracing/utils.py:19]: Great idea!

**Review by vcidst — APPROVED** [review]
This looks great! Just added a small suggestion to expand on the changelog

**Pull request merged**